src/gui/MainWindow.cpp

```diff
@@ -1284,16 +1284,40 @@ void MainWindow::processTrayIconTrigger()
     }
 }
 
+#ifndef _WIN32
+void MainWindow::show()
+{
+    m_lastShowTime = Clock::currentMiliSecondsSinceEpoch();
+    QMainWindow::show();
+}
+
+bool MainWindow::shouldHide()
+{
+    qint64 current_time = Clock::currentMiliSecondsSinceEpoch();
+
+    if (current_time - m_lastShowTime < 50)
+        return false;
+
+    return true;
+}
+
+void MainWindow::hide()
+{
+    if (shouldHide())
+        QMainWindow::hide();
+}
+#endif
+
 void MainWindow::hideWindow()
 {
     saveWindowInformation();
-#if !defined(Q_OS_LINUX) && !defined(Q_OS_MACOS)
-    // On some Linux systems, the window should NOT be minimized and hidden (i.e. not shown), at
-    // the same time (which would happen if both minimize on startup and minimize to tray are set)
-    // since otherwise it causes problems on restore as seen on issue #1595. Hiding it is enough.
-    // TODO: Add an explanation for why this is also not done on Mac (or remove the check)
-    setWindowState(windowState() | Qt::WindowMinimized);
-#endif
+    if (QGuiApplication::platformName() == "xcb") {
```

Comment:
Is this the case with Wayland as well?

Reply:
No. If and when somebody reproduces the issue on wayland it should be blacklisted as well.